SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/DynamicSubobjectsTest/DynamicSubobjectsTest.cpp

```diff
@@ -52,198 +70,251 @@ void ADynamicSubobjectsTest::PrepareTest()
-		  // AReplicatedGASTestActor.
-			AddStep(TEXT("DynamicSubobjectsTestServerMoveClient1"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
-				if (ClientOneSpawnedPawn->SetActorLocation(CharacterRemoteLocation))
+		// when in native, we need to wait for a while here - so the engine can update relevancy
+		USpatialNetDriver* CastNetDriver = Cast<USpatialNetDriver>(GetNetDriver());
+		if (!CastNetDriver)
+		{
+			AddStep(TEXT("DynamicSubobjectsTestNativeWaitABit"), FWorkerDefinition::Server(1), nullptr, nullptr, [this](float DeltaTime) {
+				StepTimer += DeltaTime;
+				if (StepTimer > 6.0f)
 				{
-					if (ClientOneSpawnedPawn->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
-					{
-						FinishStep();
-					}
+					StepTimer = 0;
+					FinishStep();
 				}
 			});
 		}
 
-		{ // Step 4 - Client 1 makes sure that the movement was correctly replicated
-			AddStep(
-				TEXT("DynamicSubobjectsTestClientCheckFirstMovement"), FWorkerDefinition::Client(1), nullptr, nullptr,
-				[this](float DeltaTime) {
-					ASpatialFunctionalTestFlowController* FlowController = GetLocalFlowController();
-					APlayerController* PlayerController = Cast<APlayerController>(FlowController->GetOwner());
-					ATestMovementCharacter* PlayerCharacter = Cast<ATestMovementCharacter>(PlayerController->GetPawn());
-
-					if (IsValid(PlayerCharacter))
-					{
-						if (PlayerCharacter->GetActorLocation().Equals(CharacterRemoteLocation, 1.0f))
-						{
-							FinishStep();
-						}
-					}
-				},
-				StepTimeLimit);
-		}
+		// Step 8 - Server increases AReplicatedGASTestActor's TestIntProperty to enable checking if the client is out of interest later.
+		AddStep(TEXT("DynamicSubobjectsTestServerIncreasesIntValue"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
+			TestActor->TestIntProperty = i + 1;
+			FinishStep();
+		});
 
-		{ // Step 5 - Server increases AReplicatedGASTestActor's TestIntProperty to enable checking if the client is out of interest later.
-			AddStep(TEXT("DynamicSubobjectsTestServerIncreasesIntValue"), FWorkerDefinition::Server(1), nullptr, [this, i]() {
-				TArray<AActor*> FoundActors;
-				UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedGASTestActor::StaticClass(), FoundActors);
-				if (FoundActors.Num() == 1)
-				{
-					TestActor = Cast<AReplicatedGASTestActor>(FoundActors[0]);
-					TestActor->TestIntProperty = i + 1;
-				}
-				if (TestActor->TestIntProperty == i + 1)
+		// Step 9 - Client 1 checks it can no longer see the AReplicatedGASTestActor
+		AddStep(
+			TEXT("DynamicSubobjectsTestClientCheckIntValueIncreased"), FWorkerDefinition::Client(1), nullptr, nullptr,
+			[this, i](float DeltaTime) {
+				RequireNotEqual_Int(TestActor->TestIntProperty, i + 1, TEXT("Check TestIntProperty didn't get replicated"));
+				StepTimer += DeltaTime;
+				if (StepTimer >= 0.5f)
 				{
 					FinishStep();
+					StepTimer = 0.0f; // reset for the next time
```

Comment:
similar to before, but I guess it is more verbose doing it the other way

Reply:
changed anyway, you had a good point